sdk-development-guidelines.md

```diff
@@ -0,0 +1,200 @@
+* We distinguish between **exported dependencies**, i.e. dependencies that consumers of our modules will automatically get in their projects as well, and **not exported dependencies**, i.e. dependencies that are completely hidden from SDK consumers.
+
+## Guidelines
+* [Project Files & Source Control Management](#project-files-&-source-control-management)
+* [Build](#build)
+* [Programming Language](#programming-language)
+* [Coding Style](#coding-style)
+* [Library Design](#library-design)
+* [Testing](#testing)
+* [Documentation](#documentation)
+* [Versioning](#versioning)
+* [Publishing](#publishing)
+* [Continuous Integration & Deployment](#continuous-integration-&-deployment)
+
+### Project files & Source Control Management
+* MUST be under source control management (SCM). RECOMMENDED source control management is git
+* If you intend your source repo to be public ensure you have received the necessary L3/L2 manager and legal department approval
+* MUST use a .gitignore configuration
+* Source code repositories SHOULD follow the naming convention `[platform]-[library name]`, e.g. `android-analytics`, `java-datastore`, `ios-inapmessaging`, `swift-utils`
+    * Rationale: encoding version information (e.g. alpha/core/stable) in the repository breaks links and integration when a module changes
+    * Exceptions: existing modules that already carry additional information in the repository name & renaming would break too many existing integrations.
+* MUST NOT restrict read access to source code repositories unnecessarily
+* For internally hosted code → add the appropriate mailing list with read access
+* For open source projects → public repo on GitHub
+* For Android <img src="images/android.png" alt="drawing" width="15"/>:
+    * SHOULD follow the standard project layout for [Android](https://developer.android.com/studio/projects) or for [Java](https://docs.gradle.org/current/userguide/java_plugin.html#sec:java_project_layout) projects respectively
+    * MUST prefix all exported resources (layout files, xml files, drawable files, color ids, dimension ids, string ids, etc.)  with <shortname> + "_" (e.g. "sug_" for a Suggestion Module) using [resourcePrefix](https://google.github.io/android-gradle-dsl/current/com.android.build.gradle.LibraryExtension.html#com.android.build.gradle.LibraryExtension:resourcePrefix) in the gradle build
+* For iOS <img src="images/ios.png" alt="drawing" width="15"/>:
+    * SHOULD follow the standard project layout for [iOS](https://developer.apple.com/documentation/xcode/managing-files-and-folders-in-your-xcode-project) projects.
+    * RECOMMEND to not have source file header comment blocks (such as the Xcode default header comment) as these easily get out of date and the information such as author/version of commits is tracked in SCM
```

Comment:
If this source file header comment block recommendation also applies to Android we can put it in the general list

Reply:
moved to general list